master/buildbot/master.py

```diff
@@ -754,20 +754,13 @@ def pollDatabaseChanges(self):
             timer.stop()
             return
 
-        while True:
-            changeid = self._last_processed_change + 1
-            chdict = yield self.db.changes.getChange(changeid)
-
-            # if there's no such change, we've reached the end and can
-            # stop polling
-            if not chdict:
-                break
-
-            change = yield changes.Change.fromChdict(self, chdict)
-
-            self._change_subs.deliver(change)
+        chdicts = yield self.db.changes.getChangesGreaterThan(self._last_processed_change)
```

Comment:
I had to change this because the old code assumes that every change in the database has `last_id + 1`, which is not always true.

Our katana databases increment ids by 3 (because or Galera multi master shenanigans), and this code was in general too naive (you can't assume that db ids are always 1 apart, you can only assume that they are incremental and unique)